java/src/test/java/org/rocksdb/BlockBasedTableConfigTest.java

```diff
@@ -95,6 +98,73 @@ public void checksumType() {
     assertThat(blockBasedTableConfig.checksumType()).isEqualTo(ChecksumType.kXXH3);
   }
 
+  @Test
+  public void jniPortal() throws Exception {
+    // Verifies that the JNI layer is correctly translating options.
+    // Since introspecting the options requires creating a database, the checks
+    // cover multiple options at the same time.
+
+    final BlockBasedTableConfig tableConfig = new BlockBasedTableConfig();
+
+    tableConfig.setIndexType(IndexType.kBinarySearch);
+    tableConfig.setDataBlockIndexType(DataBlockIndexType.kDataBlockBinarySearch);
+    tableConfig.setChecksumType(ChecksumType.kNoChecksum);
+    try (final Options options = new Options().setTableFormatConfig(tableConfig)) {
+      String opts = getOptionAsString(options);
+      assertThat(opts).contains("index_type=kBinarySearch");
+      assertThat(opts).contains("data_block_index_type=kDataBlockBinarySearch");
+      assertThat(opts).contains("checksum=kNoChecksum");
+    }
+
+    tableConfig.setIndexType(IndexType.kHashSearch);
```

Comment:
Seems this doesn't work. https://app.circleci.com/pipelines/github/sightmachine/rocksdb/27/workflows/598502ab-fda3-4932-aac8-64a4ce45dafe/jobs/131

Reply:
Sorry about that. `getOptionAsString()` wasn't calling `destroyDB`, so it was reading the old options.